File: java/src/main/java/io/github/hc1839/xbelmark/ShowMessage.java
package io.github.hc1839.xbelmark;

import javax.swing.JOptionPane;

/**
 *  Shows a message in various forms.
 */
public class ShowMessage {
  /**
   *  Shows an error message as a dialog box or, if GUI is not available, as standard error.
   *
   *  @param message
   *    Message to show.
   */
  public static void error(String message) {
    try {
      JOptionPane.showMessageDialog(null, message, "Error", JOptionPane.ERROR_MESSAGE);
    } catch (Exception e) {
      System.err.println(message);
    }
  }

  /**
   *  Shows an error message as a dialog box or, if GUI is not available, as standard error.
   *
   *  @param message
   *    Message to show.
   *
   *  @param caption
   *    Caption of the dialog box if the message is displayed in one.
   */
  public static void error(String message, String caption) {
    try {
      JOptionPane.showMessageDialog(null, message, caption, JOptionPane.ERROR_MESSAGE);
    } catch (Exception e) {
      System.err.println(message);
    }
  }
}

File: dotnet/src/main/csharp/Xbelmark/ShowMessage.cs
using System.Windows;
using System;

namespace Xbelmark;

/// <summary>
/// Shows a message in various forms.
/// </summary>
public static class ShowMessage
{
    /// <summary>
    /// Shows an error message as a dialog box in Windows, or prints an error
    /// message to the standard error in other operating systems.
    /// </summary>
    /// <param name="message">
    /// Message to show.
    /// </param>
    /// <param name="caption">
    /// Caption of the dialog box if the message is displayed in one.
    /// </param>
    public static void Error(string message, string caption = "Error")
    {
#if WINDOWS_NT
        MessageBox.Show(
            message, caption, MessageBoxButton.OK, MessageBoxImage.Error);
#else
        Console.Error.WriteLine(message);
#endif
    }
}
